selfdrive/common/params.h

```diff
@@ -17,67 +15,24 @@ enum ParamKeyType {
 
 class Params {
 public:
-  Params();
-  Params(const std::string &path);
-
+  Params(const std::string &path = {});
   bool checkKey(const std::string &key);
   ParamKeyType getKeyType(const std::string &key);
+  inline std::string getParamPath(const std::string &key) { return params_path + "/d/" + key; }
 
   // Delete a value
-  int remove(const char *key);
-  inline int remove(const std::string &key) {
-    return remove (key.c_str());
-  }
+  int remove(const std::string &key);
   void clearAll(ParamKeyType type);
 
-  // read all values
-  std::map<std::string, std::string> readAll();
-
   // helpers for reading values
-  std::string get(const char *key, bool block = false);
-
-  inline std::string get(const std::string &key, bool block = false) {
-    return get(key.c_str(), block);
-  }
-
-  inline std::string getParamsPath() {
-    return params_path;
-  }
-
-  inline std::string getParamPath(std::string key) {
-    return params_path + "/d/" + key;
-  }
-
-  template <class T>
-  std::optional<T> get(const char *key, bool block = false) {
-    std::istringstream iss(get(key, block));
-    T value{};
-    iss >> value;
-    return iss.fail() ? std::nullopt : std::optional(value);
-  }
-
-  inline bool getBool(const std::string &key) {
-    return getBool(key.c_str());
-  }
-
-  inline bool getBool(const char *key) {
-    return get(key) == "1";
-  }
+  std::string get(const std::string &key, bool block = false);
+  inline bool getBool(const std::string &key) { return get(key) == "1"; }
+  std::map<std::string, std::string> readAll();
 
   // helpers for writing values
-  int put(const char* key, const char* val, size_t value_size);
-
-  inline int put(const std::string &key, const std::string &val) {
-    return put(key.c_str(), val.data(), val.size());
-  }
-
-  inline int putBool(const char *key, bool val) {
-    return put(key, val ? "1" : "0", 1);
-  }
-
-  inline int putBool(const std::string &key, bool val) {
-    return putBool(key.c_str(), val);
-  }
+  int put(const char *key, const char *val, size_t value_size);
+  inline int put(const std::string &key, const std::string &val) { return put(key.c_str(), val.data(), val.size()); }
```

Comment:
let's leave these expanded. no need to save lines like this

Reply:
ok :smile: